src/pixelator/pixeldataset/precomputed_layouts.py

```diff
@@ -253,81 +387,94 @@ def data():
         return PreComputedLayouts.create_empty()
 
 
-# TODO The code below this point is not yet tested, and should be considered
-# experimental. It is likely to change in the future. It will be exposed as
-# public to this method in the future.
-
-
 def _zero_fill_missing_markers(dataframe: pd.DataFrame, all_markers) -> pd.DataFrame:
     missing_markers = all_markers - set(dataframe.columns.to_list())
     for marker in missing_markers:
         dataframe[marker] = 0
     return dataframe
 
 
-def _get_layout(
+def _compute_layouts(
     edgelist: pl.DataFrame,
-    add_node_marker_counts,
-    layout_algorithm,
-    all_markers,
+    add_node_marker_counts: bool,
+    layout_algorithms: list[SupportedLayoutAlgorithm] | SupportedLayoutAlgorithm,
+    all_markers: list[str],
 ):
+    if isinstance(layout_algorithms, str):
```

Comment:
As we have typed this `layout_algorithms`, is this really a `str`or a `SupportedLayoutAlgorithm`?

Reply:
The way I understand literals they are only a type checking mechanism, at runtime they are just their concert types, i.e. `str` in this case. You can read more about Literal types here: https://typing.readthedocs.io/en/latest/spec/literal.html#literal-types